File: src/main/java/sk/hudak/prco/utils/PatternUtils.java
package sk.hudak.prco.utils;

import lombok.NoArgsConstructor;

import java.util.StringJoiner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@NoArgsConstructor
public final class PatternUtils {

    public static final String NUMBER_AT_LEAST_ONE = "[0-9]{1,}";
    public static final String SPACE = " ";

    public static Matcher createMatcher(String value, String... groups) {
        return cratePattern(groups).matcher(value);
    }

    public static Pattern cratePattern(String... groups) {
        StringJoiner stringJoiner = new StringJoiner(")(");
        for (String group : groups) {
            stringJoiner.add(group);
        }
        return Pattern.compile("(" + stringJoiner.toString() + ")");
    }
}

File: src/main/java/sk/hudak/prco/utils/PatternUtils.kt
package sk.hudak.prco.utils

import java.util.*
import java.util.regex.Matcher
import java.util.regex.Pattern


class PatternUtils {

    companion object {

        const val NUMBER_AT_LEAST_ONE = "[0-9]{1,}"
        const val SPACE = " "

        @JvmStatic // tato anotacia tu musi byt ak chcem aby dany kotlin kod bolo mozne volat aj priamo z javy !
        fun createMatcher(value: String, vararg groups: String): Matcher {
            return cratePattern(*groups).matcher(value)
        }

        @JvmStatic
        fun cratePattern(vararg groups: String): Pattern {
            val stringJoiner = StringJoiner(")(")
            for (group in groups) {
                stringJoiner.add(group)
            }
            return Pattern.compile("($stringJoiner)")
        }
    }

}
